### PR TITLE
Clean up React warnings and errors

### PR DESCRIPTION
**Pull request by mgwalker**
Adds keys, fixes prop types, and makes sure CSS class names aren't booleans.

There are three categories of outstanding errors/warnings:

- Prop type error passing the `fieldRef` prop into the design system `TextField` component. The component only wants a function, but React accepts both functions and objects. Objects are the newer, more preferred approach, so I've stuck with that. We can open an issue with the design system to update their prop types.
- There are warnings about `componentWillMount` and `componentWillReceiveProps`, both deprecated lifecycle methods. We're not using these in our code anymore, so those warnings are coming from our dependencies. We can't really do anything about them.
- Nearly 300 warnings due to deprecated events being subscribed by the current rich text editor we're using. It should continue to work fine, but those deprecated events could prevent us from upgrading to a more recent version of React at some point in the future (e.g., React 17 or 18).

### This pull request changes...

- fixes most of the React warnings and errors caused by our app

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author

### CODE REVIEW

File: web/src/containers/BudgetSummary.js [hunk 1]
```diff
@@ -178,11 +159,13 @@ const BudgetSummary = ({ activities, data, years }) => (
           return (
             <tr
               key={ffy}
-              className={ffy === "total" && "budget-table--row__highlight budget-table--total"}
```
Comment (mgwalker) [inline web/src/containers/BudgetSummary.js]: This results in `className` being either `"budget-table--row__highlight budget-table--total"` or `false`. It turns out that React shouts at you if your `className` isn't a string. Who knew?

File: web/src/containers/activity/ContractorResource/ContractorResourceReview.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ ContractorResourceReview.propTypes = {
     name: PropTypes.string,
     start: PropTypes.string,
     totalCost: PropTypes.number,
-    years: PropTypes.array
```
Comment (mgwalker) [inline web/src/containers/activity/ContractorResource/ContractorResourceReview.js]: This was never an array. 🤷‍♂

File: web/src/containers/activity/StatePersonnel.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ const StatePersonnel = ({ activityIndex, add, personnel, remove }) => {
 };
 
 StatePersonnel.propTypes = {
-  activityIndex: PropTypes.string.isRequired,
```
Comment (mgwalker) [inline web/src/containers/activity/StatePersonnel.js]: And this was never a string. 🤷‍♂ 🤷‍♂

File: web/src/components/Header.js [hunk 4]
```diff
@@ -126,11 +126,12 @@ Header.propTypes = {
   authenticated: PropTypes.bool.isRequired,
   currentUser: PropTypes.object,
   isAdmin: PropTypes.bool.isRequired,
-  ariaExpanded: PropTypes.bool.isRequired,
+  ariaExpanded: PropTypes.bool,
   showSiteTitle: PropTypes.bool.isRequired
 };
 
 Header.defaultProps = {
+  ariaExpanded: false,
```
Comment (hbillings) [inline web/src/components/Header.js]: 👌

File: web/src/components/NumberField.js [hunk 5]
```diff
@@ -19,6 +19,9 @@ const NumberField = ({ fieldRef, ...props }) => {
     };
   }, []);
 
+  // A bug in the CMS design system says the fieldRef prop must be a function,
+  // but React also accepts objects. Objects are the newer approach, so we'll
+  // stick with that and just accept the prop types warning for now.
```
Comment (hbillings) [inline web/src/components/NumberField.js]: Thanks for documenting this here!

File: web/src/containers/BudgetSummary.js (hunk 1 above)
Comment (hbillings) [reply web/src/containers/BudgetSummary.js]: Are you saying that the component... has no class? ;-)

**Pull request merged**